### PR TITLE
Support the generation of JMeter performance testing scripts

### PR DESCRIPTION
**Pull request by Rancho-7**
### What does this PR do? 
Generate .jmx files for jmeter,import the file in jmeter to use.
Support the generation of JMeter performance testing scripts:https://github.com/TongchengOpenSource/smart-doc/issues/653

### Checklist: 清单：
● Supports Thread Groups and Samplers for each specific interface
image1

● Parameters that support get requests are presented in "Parameters"
image2

● Supports form-data type parameters, which are presented in "Parameters"
image3
image4

● Supports data in json format, presented in "Body Data"
image5

### CODE REVIEW

File: src/main/resources/template/JmxApiDoc.jmx [hunk 1]
```diff
@@ -0,0 +1,161 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jmeterTestPlan version="1.2" properties="5.0" jmeter="5.4.3">
+  <hashTree>
+    <TestPlan guiclass="TestPlanGui" testclass="TestPlan" testname="${projectName}" enabled="true">
+      <stringProp name="TestPlan.comments"></stringProp>
+      <boolProp name="TestPlan.functional_mode">false</boolProp>
+      <boolProp name="TestPlan.tearDown_on_shutdown">true</boolProp>
+      <boolProp name="TestPlan.serialize_threadgroups">false</boolProp>
+      <elementProp name="TestPlan.user_defined_variables" elementType="Arguments" guiclass="ArgumentsPanel" testclass="Arguments" testname="用户定义的变量" enabled="true">
```
Comment (shalousun) [inline src/main/resources/template/JmxApiDoc.jmx]: It is recommended to do internationalization display in Chinese and English according to the configuration option value of DocLanguage.

**Comment by shalousun** [comment]
@Rancho-7  rebase comment messages,  you can reference https://smart-doc-group.github.io/#/zh-cn/community/pull-request-process.md

**Comment by shalousun** [comment]
The smart-doc-maven-plugin and smart-doc-gradle plugin also need to add support for JMeter.

**Comment by Rancho-7** [comment]
finished




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年11月27日(星期一) 上午10:46
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [TongchengOpenSource/smart-doc] Support the generation of JMeter performance testing scripts (PR #668)





 
@Rancho-7  rebase comment messages,  you can reference https://smart-doc-group.github.io/#/zh-cn/community/pull-request-process.md

File: src/main/java/com/ly/doc/builder/JmxDocBuilder.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import com.power.common.util.DateTimeUtil;
+import com.thoughtworks.qdox.JavaProjectBuilder;
+
+import java.util.List;
+
+/**
+ * used to generate jmx file for Jmeter
+ * @author Lansg
+ */
+public class JmxDocBuilder {
```
Comment (shalousun) [inline src/main/java/com/ly/doc/builder/JmxDocBuilder.java]: Modify JmxDocBuilder to JMeterBuilder.

File: src/main/java/com/ly/doc/constants/DocGlobalConstants.java [hunk 3]
```diff
@@ -42,6 +42,8 @@ public interface DocGlobalConstants {
     String RANDOM_MOCK = "randomMock";
     String API_DOC_MD_TPL = "ApiDoc.md";
 
+    String API_DOC_JMX_TPL = "JmxApiDoc.jmx";
```
Comment (shalousun) [inline src/main/java/com/ly/doc/constants/DocGlobalConstants.java]: Modify to   JMETER_TPL = "JMeter.jmx";

File: src/main/resources/template/JmxApiDoc.jmx (hunk 1 above)
Comment (Rancho-7) [reply src/main/resources/template/JmxApiDoc.jmx]: finished

**Review by shalousun — APPROVED** [review]

**Comment by Rancho-7** [comment]
finished



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年11月27日(星期一) 下午2:33
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [TongchengOpenSource/smart-doc] Support the generation of JMeter performance testing scripts (PR #668)





 
The smart-doc-maven-plugin and smart-doc-gradle plugin also need to add support for JMeter.

**Pull request merged**